### DISCUSSION TITLE
part-ii/2018-101/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#166] [Question comments]
Part II 2018 Number Theory: Paper 3, Section I, G 

https://questions.tripos.org/part-ii/2018-101/

**Comment by dropdbassett** [comment]
Mostly bookwork. My solution for the final part was somewhat strange, so I'd appreciate if someone has a more sensible solution:
<details><summary>What is f if g is the identity map?</summary>

Note from the Möbius inversion formula proven, there is a unique such $f$. We show that $f(n) = \varphi(n)$ is a solution: consider the fractions $\frac{1}{n}, \dots, \frac{n}{n}$, written in reduced form. Clearly any denominator of these is a positive divisor of $n$, and for $d \mid n$, the fractions written with denominator $d$ are precisely $\frac{k}{d}$ for $1 \le k < d$ with $\gcd(k, d) = 1$. Thus, as desired,

$$\sum\limits_{d \mid n} \varphi(d) = n.$$
</details>

**Reply by callumhobbis** [reply]
<details>
<summary>What is f(n) if g(n)=n?</summary>

I did a similar thing, recalling $\sum_{d\mid n}\varphi(d)=n$ (and proving it since I would be worried stating it isn't enough) and so that must be the solution by uniqueness. The proof you gave is in lectures (lecture 2) as an alternative proof! So I wouldn't say it's not sensible.

</details>

**Reply by dropdbassett** [reply]
Oh sure - I'd forgotten that it was given in lectures :)
